.azure-pipelines/gpu-tests.yml

```diff
@@ -108,8 +108,6 @@ jobs:
         bash pl_examples/run_examples.sh --trainer.gpus=1
         bash pl_examples/run_examples.sh --trainer.gpus=2 --trainer.strategy=ddp
         bash pl_examples/run_examples.sh --trainer.gpus=2 --trainer.strategy=ddp --trainer.precision=16
-        bash pl_examples/run_examples.sh --trainer.gpus=2 --trainer.strategy=dp
-        bash pl_examples/run_examples.sh --trainer.gpus=2 --trainer.strategy=dp --trainer.precision=16
```

Comment:
removed, as this PR adds the torchmetrics to some of the examples and this is incompatible with DP. 

Considered options:
- switch to ddp_spawn: not working, because at least one example is returning predictions: 
`predictions = cli.trainer.predict(ckpt_path="best", datamodule=cli.datamodule)`
and predictions will be empty here, causing an error.
- hardcoding dp/ddp_spawn into the examples: not really working, because it would error when trying to run the them on CPU :(